### DISCUSSION TITLE
Setup CreateVkSurface handler

### DISCUSSION THREAD

**Discussion by lksrb** [ocornut/imgui#5871] [New Users Build/Link/Run issues ONLY!]
Hi, 
I've been trying to implement Win32-Vulkan ImGui backend into my custom engine and I ran into assert ```Platform needs to setup the CreateVkSurface handler```.

How do I properly setup this handler?

Thanks for any help :)

**Comment by PathogenDavid** [comment]
The Vulkan+Win32+multi-viewport combination is not supported out of the box.

The `CreateVkSurface` handler is typically set up by the platform backend and is used by `ImGui_ImplVulkan_CreateWindow` to initialize the Vulkan surface for the window, but `imgui_impl_win32` does not implement this function.

You can manually implement it yourself by assigning a function to `ImGui::GetPlatformIO().Platform_CreateVkSurface` before calling `ImGui_ImplVulkan_Init`. Unfortunately I'm not aware of any examples on how to do this, but presumably you should have most of what you need in your engine already.

You can access the HWND the surface is being created for via `ImGuiViewport::PlatformHandleRaw`.

Here's the implementation for `imgui_impl_glfw` in case it's useful:

https://github.com/ocornut/imgui/blob/5728f2e15fe5354c84254cddcf1f01713c10374a/backends/imgui_impl_glfw.cpp#L1132-L1140

and `imgui_impl_sdl`:

https://github.com/ocornut/imgui/blob/5728f2e15fe5354c84254cddcf1f01713c10374a/backends/imgui_impl_sdl.cpp#L901-L907

**Reply by lksrb** [reply]
Ah I see, thank you. :)
